packages/docusaurus-plugin-content-showcase/src/lifecycle/contentLoaded.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import type {ShowcaseItems} from '@docusaurus/plugin-content-showcase';
+import type {PluginContentLoadedActions} from '@docusaurus/types';
+
+export async function processContentLoaded({
+  content,
+  routeBasePath,
+  addRoute,
+  createData,
+}: {
+  content: ShowcaseItems;
+  routeBasePath: string;
+  addRoute: PluginContentLoadedActions['addRoute'];
+  createData: PluginContentLoadedActions['createData'];
+}): Promise<void> {
+  const showcaseAllData = await createData(
+    'showcaseAll.json',
+    JSON.stringify(content.items),
+  );
+
+  addRoute({
+    path: routeBasePath,
+    component: '@theme/Showcase',
+    modules: {
+      content: showcaseAllData,
+    },
+    exact: true,
+  });
```

Comment:
If you update your PR with main, you should be able to use the new route props feature (https://github.com/facebook/docusaurus/pull/10042)

```suggestion
  addRoute({
    path: routeBasePath,
    component: '@theme/Showcase',
    exact: true,
    props: {
      items: content.items,
    },
  });
```

Also probably better to name the prop "items" since it's an array of showcase items, and not directly the loaded content object